### PR TITLE
Update dg nfr

### PR DESCRIPTION
**Pull request by seantham21**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -520,23 +520,31 @@ Use case ends.
...
-13. The system should provide a way to export and import data in a standard format (e.g., CSV) to allow for easy backups and data transfer between devices.
-14. The system should provide a way to archive old data to reduce clutter and improve performance, ensuring that archived data can be easily retrieved if needed.
-15. New commands or features should be easily added or modified without impacting the existing system. This includes the ability to update or patch the app efficiently.
-16. Since the app is intended to be an offline tool, it should function fully without any internet connection. All features, including scheduling and contact management, must be operable offline.
-17. The application should be lightweight, with minimal CPU and memory usage, allowing it to run smoothly on a wide variety of machines, including low-spec systems.
-18. The system should provide a user-friendly and intuitive interface with clear instructions and feedback to guide users who are not IT-savvy in using the command-line interface effectively.
+2. The software should work without requiring an installer.
+3. The software should not depend on your own remote server.
```
Comment (wrjgold) [inline docs/DeveloperGuide.md:524]: Should change from "your own" to "a"

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -520,23 +520,31 @@ Use case ends.
...
-17. The application should be lightweight, with minimal CPU and memory usage, allowing it to run smoothly on a wide variety of machines, including low-spec systems.
-18. The system should provide a user-friendly and intuitive interface with clear instructions and feedback to guide users who are not IT-savvy in using the command-line interface effectively.
+2. The software should work without requiring an installer.
+3. The software should not depend on your own remote server. 
+4. Since the app is intended to be an offline tool, it should function fully without any internet connection.
+5. The application should be packaged into a single JAR file.
+6. The file sizes of the deliverables should be reasonable and not exceed the limits given below. 
+   - Product (i.e., the JAR/ZIP file): 100MB
+   - Documents (i.e., PDF files): 15MB/file
+7. The final JAR/PDF files cannot be bloated unnecessarily.
```
Comment (wrjgold) [inline docs/DeveloperGuide.md:530]: Not sure what's the meaning / metric for this

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -520,23 +520,31 @@ Use case ends.
...
+8. The developer guide and user guide should be PDF-friendly. Don't use expandable panels, embedded videos, animated GIFs etc.
+9. The use of third-party libraries/frameworks/services are not allowed unless they are approved by the teaching team.
```
Comment (wrjgold) [inline docs/DeveloperGuide.md:532]: remove "teaching team", change reason into something like "to improve security and stability of the application"

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -520,23 +520,31 @@ Use case ends.
...
+10. The software should follow the Object-oriented paradigm primarily.
```
Comment (wrjgold) [inline docs/DeveloperGuide.md:533]: I feel can remove

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -520,23 +520,31 @@ Use case ends.
...
+11. A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse.
```
Comment (wrjgold) [inline docs/DeveloperGuide.md:534]: Phrase it into something more related to our product? "Our software should enable talent managers with above average..."

File: docs/DeveloperGuide.md [hunk 6]
```diff
@@ -520,23 +520,31 @@ Use case ends.
...
+12. The application should prioritise one-shot commands over multi-step commands as they are faster. If a multi-step command is provided to help new users, a one-shot equivalent should also be provided for regular/expert users.
+13. The application should respond to user commands within 0.5 seconds, ensuring fast interactions.
+14. Regular tasks (e.g., adding a contact, listing events) should be executable within a few milliseconds to maintain efficiency and not interrupt the user’s workflow.
+15. The system should efficiently handle a large database of contacts and events (e.g., 1,000+ entries), allowing for fast searches, retrievals, and modifications without performance degradation.
+16. The data should be stored locally and should be in a human editable text file.
+17. The data should not be stored using a Database Management System.
+18. The data should be able to be loaded using a data file if the network is down.
+19. The app should allow password protection and encryption options to ensure that only authorized users can access or modify sensitive data.
```
Comment (wrjgold) [inline docs/DeveloperGuide.md:542]: Should remove this

File: docs/DeveloperGuide.md [hunk 7]
```diff
@@ -520,23 +520,31 @@ Use case ends.
...
-15. New commands or features should be easily added or modified without impacting the existing system. This includes the ability to update or patch the app efficiently.
-16. Since the app is intended to be an offline tool, it should function fully without any internet connection. All features, including scheduling and contact management, must be operable offline.
-17. The application should be lightweight, with minimal CPU and memory usage, allowing it to run smoothly on a wide variety of machines, including low-spec systems.
-18. The system should provide a user-friendly and intuitive interface with clear instructions and feedback to guide users who are not IT-savvy in using the command-line interface effectively.
+2. The software should work without requiring an installer.
+3. The software should not depend on your own remote server. 
+4. Since the app is intended to be an offline tool, it should function fully without any internet connection.
+5. The application should be packaged into a single JAR file.
+6. The file sizes of the deliverables should be reasonable and not exceed the limits given below.
```
Comment (fusin02) [inline docs/DeveloperGuide.md:527]: Maybe just say file size (not deliverables)

File: docs/DeveloperGuide.md [hunk 8]
```diff
@@ -520,23 +520,31 @@ Use case ends.
...
+   - Product (i.e., the JAR/ZIP file): 100MB
+   - Documents (i.e., PDF files): 15MB/file
+7. The final JAR/PDF files cannot be bloated unnecessarily.
+8. The developer guide and user guide should be PDF-friendly. Don't use expandable panels, embedded videos, animated GIFs etc.
+9. The use of third-party libraries/frameworks/services are not allowed unless they are approved by the teaching team.
+10. The software should follow the Object-oriented paradigm primarily.
+11. A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse.
+12. The application should prioritise one-shot commands over multi-step commands as they are faster. If a multi-step command is provided to help new users, a one-shot equivalent should also be provided for regular/expert users.
+13. The application should respond to user commands within 0.5 seconds, ensuring fast interactions.
+14. Regular tasks (e.g., adding a contact, listing events) should be executable within a few milliseconds to maintain efficiency and not interrupt the user’s workflow.
```
Comment (fusin02) [inline docs/DeveloperGuide.md:537]: 13 and 14 i think might be a bit overlappy

**Review by fusin02 — APPROVED** [review]

**Pull request merged**